docs/analyzer/adding_recognizers.md

```diff
@@ -0,0 +1,247 @@
+# Supporting new types of PII entities
+
+Presidio can be extended to support new types of PII entities, and to support additional languages.
+
+## Table of contents
+
+- [Introduction to recognizer development](#introduction-to-recognizer-development)
+- [Types of recognizer classes in Presidio](#types-of-recognizer-classes-in-presidio)
+- [Extending the analyzer for additional PII entities](#extending-the-analyzer-for-additional-pii-entities)
+- [Extending to additional languages](#extending-to-additional-languages)
+
+## Introduction to recognizer development
+
+Entity recognizers are Python objects capable of detecting one or more entities in a specific language.
+In order to extend Presidio's detection capabilities to new types of PII entities,
+these `EntityRecognizer` objects should be added to the existing list of recognizers.
+
+## Types of recognizer classes in Presidio
+
+The following class diagram shows the different types of recognizer families Presidio contains.
+
+![Recognizers class diagram](../assets/recognizers_class_diagram.png)
+
+- The `EntityRecognizer` is an abstract class for all recognizers.
+- The `RemoteRecognizer` is an abstract class for calling external PII detectors.
+See more info [here](#creating-a-remote-recognizer).
+- The `LocalRecognizer` is an abstract class for all recognizers running within the Presidio-analyzer process.
+- The `PatternRecognizer` is an class for supporting regex and deny-list based recognition logic,
+including validation (e.g., with checksum) and context support. See an example [here](#simple-example).
+
+## Extending the analyzer for additional PII entities
+
+First, a class based on `EntityRecognizer` needs to be created.
```

Comment:
Suggestion: another option, that might increase readability

1. Create a new class based on `EntityRecognizer`
2. Add the new recognizer to the recognizer registry so that the `AnalyzerEngine` can use the new recognizer during analysis.